Validation/RecoParticleFlow/plugins/PFCaloGPUComparisonTask.cc

```diff
@@ -0,0 +1,170 @@
+#include "DQMServices/Core/interface/DQMEDAnalyzer.h"
+#include "DQMServices/Core/interface/DQMStore.h"
+#include "DataFormats/CaloRecHit/interface/CaloCluster.h"
+#include "DataFormats/CaloRecHit/interface/CaloClusterFwd.h"
+#include "DataFormats/CaloTowers/interface/CaloTowerCollection.h"
+#include "DataFormats/CaloTowers/interface/CaloTowerDetId.h"
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/DetId/interface/DetId.h"
+#include "DataFormats/EcalDetId/interface/EcalSubdetector.h"
+#include "DataFormats/HcalDetId/interface/HcalDetId.h"
+#include "DataFormats/HcalDetId/interface/HcalSubdetector.h"
+#include "DataFormats/Math/interface/Vector3D.h"
+#include "DataFormats/Math/interface/deltaR.h"
+#include "DataFormats/ParticleFlowCandidate/interface/PFCandidate.h"
+#include "DataFormats/ParticleFlowReco/interface/PFBlock.h"
+#include "DataFormats/ParticleFlowReco/interface/PFBlockElementCluster.h"
+#include "DataFormats/ParticleFlowReco/interface/PFBlockElementTrack.h"
+#include "DataFormats/ParticleFlowReco/interface/PFCluster.h"
+#include "DataFormats/ParticleFlowReco/interface/PFClusterFwd.h"
+#include "DataFormats/ParticleFlowReco/interface/PFLayer.h"
+#include "DataFormats/ParticleFlowReco/interface/PFRecHit.h"
+#include "DataFormats/ParticleFlowReco/interface/PFRecHitFraction.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/PluginManager/interface/ModuleDef.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "SimDataFormats/GeneratorProducts/interface/HepMCProduct.h"
+
+#include <algorithm>
+#include <cmath>
+#include <iostream>
+#include <string>
+#include <utility>
+#include <vector>
+
+#ifdef PFLOW_DEBUG
+#define LOGVERB(x) edm::LogVerbatim(x)
+#else
+#define LOGVERB(x) LogTrace(x)
+#endif
+
+class PFCaloGPUComparisonTask : public DQMEDAnalyzer {
+public:
+  PFCaloGPUComparisonTask(edm::ParameterSet const& conf);
+  ~PFCaloGPUComparisonTask() override = default;
+  void analyze(edm::Event const& e, edm::EventSetup const& c) override;
+  void bookHistograms(DQMStore::IBooker&, edm::Run const&, edm::EventSetup const&) override;
+
+private:
+  edm::EDGetTokenT<reco::PFClusterCollection> pfClusterTok_ref_;
+  edm::EDGetTokenT<reco::PFClusterCollection> pfClusterTok_target_;
+
+  MonitorElement* pfCluster_Multiplicity_GPUvsCPU_;
+  MonitorElement* pfCluster_Energy_GPUvsCPU_;
+  MonitorElement* pfCluster_RecHitMultiplicity_GPUvsCPU_;
+  MonitorElement* pfCluster_Layer_GPUvsCPU_;
+  MonitorElement* pfCluster_Depth_GPUvsCPU_;
+  MonitorElement* pfCluster_Eta_GPUvsCPU_;
+  MonitorElement* pfCluster_Phi_GPUvsCPU_;
+
+  std::string pfCaloGPUCompDir;
```

Comment:
```suggestion
  std::string pfCaloGPUCompDir_;
```

sorry I missed this earlier